sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,1033 @@
+    ;
+
+    if (!StringUtils.isBlank(index)
+        && !StringUtils.isBlank(type)
+        && !StringUtils.isBlank(identifier)
+        && data != null) {
+      Map<String, String> mappedIndexAndType =
+          ElasticSearchHelper.getMappedIndexAndType(index, type);
+      UpdateRequest updateRequest =
+          new UpdateRequest(
+                  mappedIndexAndType.get(JsonKey.INDEX),
+                  mappedIndexAndType.get(JsonKey.TYPE),
+                  identifier)
+              .doc(data);
+
+      ActionListener<UpdateResponse> listener =
+          new ActionListener<UpdateResponse>() {
+            @Override
+            public void onResponse(UpdateResponse updateResponse) {
+              promise.success(true);
+              if (updateResponse.getResult() == DocWriteResponse.Result.CREATED) {
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  Success with upsert for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              } else if (updateResponse.getResult() == DocWriteResponse.Result.UPDATED) {
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  Success for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              } else if (updateResponse.getResult() == DocWriteResponse.Result.NOOP) {
+                promise.success(false);
+                ProjectLogger.log(
+                    "ElasticSearchUtilRest updateData  falied for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              }
+
+              long stopTime = System.currentTimeMillis();
+              long elapsedTime = stopTime - startTime;
+              ProjectLogger.log(
+                  "ElasticSearchUtilRest updateData method end at =="
+                      + stopTime
+                      + " for Type "
+                      + type
+                      + " ,Total time elapsed = "
+                      + elapsedTime,
+                  LoggerEnum.PERF_LOG);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              ProjectLogger.log(
+                  "ElasticSearchUtilRest : updateData exception occured:" + e.getMessage(),
+                  LoggerEnum.ERROR.name());
+              promise.failure(e);
+            }
+          };
+      ConnectionManager.getRestClient().updateAsync(updateRequest, listener);
+
+    } else {
+      ProjectLogger.log(
```

Comment:
Here we need to throw failure client error, becuase data was not provided by caller as expected.

Reply:
Done